hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/contract/opcodes/PushZeroOperationSuite.java

```diff
@@ -0,0 +1,118 @@
+import static com.hedera.services.bdd.spec.transactions.TxnVerbs.cryptoCreate;
+import static com.hedera.services.bdd.spec.transactions.TxnVerbs.uploadInitCode;
+import static com.hedera.services.bdd.spec.utilops.UtilVerbs.overriding;
+import static com.hedera.services.bdd.spec.utilops.UtilVerbs.sourcing;
+
+import com.hedera.services.bdd.spec.HapiSpec;
+import com.hedera.services.bdd.suites.HapiSuite;
+import com.hederahashgraph.api.proto.java.ResponseCodeEnum;
+import java.math.BigInteger;
+import java.util.List;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+public class PushZeroOperationSuite extends HapiSuite {
+    private static final Logger log = LogManager.getLogger(PushZeroOperationSuite.class);
+    private static final long GAS_TO_OFFER = 400_000L;
+    private static final String CONTRACT = "OpcodesContract";
+    private static final String BOB = "bob";
+
+    private static final String OP_PUSH_ZERO = "opPush0";
+
+    public static final String CONTRACTS_DYNAMIC_EVM_VERSION = "contracts.evm.version.dynamic";
+    public static final String CONTRACTS_EVM_VERSION = "contracts.evm.version";
+
+    public static final String EVM_VERSION_0_34 = "v0.34";
+    public static final String EVM_VERSION_0_38 = "v0.38";
+
+    public static void main(String... args) {
+        new PushZeroOperationSuite().runSuiteSync();
+    }
+
+    @Override
+    public List<HapiSpec> getSpecsInSuite() {
+        return allOf(positiveSpecs(), negativeSpecs());
+    }
+
+    List<HapiSpec> negativeSpecs() {
+        return List.of();
+    }
+
+    List<HapiSpec> positiveSpecs() {
+        return List.of(pushZeroHappyPathWorks(), pushZeroDisabledInV034());
+    }
+
+    private HapiSpec pushZeroHappyPathWorks() {
+        final var pushZeroContract = CONTRACT;
+        final var pushResult = "pushResult";
+        return defaultHapiSpec("prngPrecompileHappyPathWorks")
+                .given(
+                        overriding(CONTRACTS_DYNAMIC_EVM_VERSION, TRUE_VALUE),
+                        overriding(CONTRACTS_EVM_VERSION, EVM_VERSION_0_38),
+                        cryptoCreate(BOB),
+                        uploadInitCode(pushZeroContract),
+                        contractCreate(pushZeroContract))
+                .when(sourcing(() -> contractCall(pushZeroContract, OP_PUSH_ZERO)
+                        .gas(GAS_TO_OFFER)
+                        .payingWith(BOB)
+                        .via(pushResult)
+                        .logged()))
+                .then(getTxnRecord(pushResult)
+                        .hasPriority(recordWith()
+                                .contractCallResult(resultWith()
+                                        .resultViaFunctionName(
+                                                OP_PUSH_ZERO,
+                                                pushZeroContract,
+                                                isLiteralResult((new Object[] {BigInteger.valueOf(0x5f)})))))
```

Comment:
I obviously don't know how this test suite works but this seems odd to me: Why is the literal result of a `PUSH0` test a `0x5F` instead of a `0x00`?  How does this test work?  Oh wait, I see now `0x5F` is the opcode for `PUSH0` - but then why doesn't this test actually test a `0` was pushed on the stack?  Or is that not the purpose of this test - opcode correct execution is tested elsewhere (like in besu itself) and this test suite is ... just to make sure the opcode actually exists?

(I suppose it is too much to ask but is there a doc somewhere that describes what these tests under `bdd.suites.contract` are for?  I'm not asking you to write one now, but did anyone write one ever?  Where is it?)

Reply:
There is a whole lot of yak shaving that goes on in an EET.  I can't change that, the alternative is to not test that PUSH0 is activated in 0.38 and not activated in 0.34.

Per the conventions the solidity is kept in a different file, (referenced via `pushZeroContract` variable, via static variable `CONTRACT`, going to `"OpcodesContract"`, which resolves to the [OpcodeContract](https://github.com/hashgraph/hedera-services/tree/4a89d5a32a7292709f79f5509c6e26d2cb20ccc9/hedera-node/test-clients/src/main/resource/contract/contracts/OpcodesContract) directory in the resources fork of the test fork of the test-clients module.

The `OP_PUSH_ZERO` static field maps to `"opPush0"` which corresponds to the [`opPush0`](https://github.com/hashgraph/hedera-services/blob/4a89d5a32a7292709f79f5509c6e26d2cb20ccc9/hedera-node/test-clients/src/main/resource/contract/contracts/OpcodesContract/OpcodesContract.sol#L37) function in the corresponding smart contract.  I'll address how this is compiled in another comment, but the commented out line is actually what gets executed: a push0, a push1  pushing 0x5f, then an add, then returning that value.  A push 0 returning a zero could be mistaken for a do-nothing function success, so I had to do something.

This test is duplicated twice (this function and the one below), with different values in the `overriding` section in the when clause.  One activating the 0.38 VM and one the 0.34.  The assert results are different because the VM code should revert if not activated.

Reply:
Resolving to unblock. 
We should consider adding comments in follow up